### PR TITLE
docs: basic auth example use https

### PR DESCRIPTION
**Pull request by phantooom**
**What type of PR is this?**
* "docs: basic auth example use https"

https://github.com/envoyproxy/gateway/issues/2792

### CODE REVIEW

File: site/content/en/latest/user/basic-auth.md [hunk 1]
```diff
@@ -21,6 +21,55 @@ Envoy Gateway uses [.htpasswd][.htpasswd] format to store the username-password
 The file must be stored in a kubernetes secret and referenced in the [SecurityPolicy][SecurityPolicy] configuration. 
 The secret is an Opaque secret, and the username-password pairs must be stored in the key ".htpasswd".
 
+Note: We recommend users to enable HTTPS when using Basic Authentication.
+
```
Comment (arkodg) [inline site/content/en/latest/user/basic-auth.md:25]: nice ! only one nit: can we move this diff before L28, so the `.htpasswd` sentence is below this ?

Comment (phantooom) [reply site/content/en/latest/user/basic-auth.md:25]: 👌

**Comment by zirain** [comment]
EG use squash merge mode, please kindly don't force push every time for better review.

**Comment by phantooom** [comment]
> EG use squash merge mode, please kindly don't force push every time for better review.

ok

**Comment by Xunzhuo** [comment]
Can you sync this to new path @phantooom ?

File: site/content/en/latest/user/basic-auth.md
Comment (zirain) [inline site/content/en/latest/user/basic-auth.md:1]: should remove this file?

File: site/content/en/latest/user/security/basic-auth.md [hunk 2]
```diff
@@ -21,9 +21,58 @@ Envoy Gateway uses [.htpasswd][.htpasswd] format to store the username-password
...
+        }],
+      },
+    },
+}]'
+```
+
 ### Create a .htpasswd file
 First, create a [.htpasswd][.htpasswd] file with the username and password you want to use for authentication. 
 
+Note: We recommend users to enable HTTPS when using Basic Authentication.
```
Comment (zhaohuabing) [inline site/content/en/latest/user/security/basic-auth.md:74]:
```suggestion
Note: Please always use HTTPS with Basic Authentication. This prevents credentials from being transmitted in plain text.
```

**Review by zhaohuabing — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**